src/main/java/org/folio/dao/transactions/BatchTransactionPostgresDAO.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.dao.transactions;
+
+import io.vertx.core.Future;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.CriterionBuilder;
+import org.folio.rest.persist.DBConn;
+import org.folio.rest.persist.interfaces.Results;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BatchTransactionPostgresDAO implements BatchTransactionDAO {
+  private static final Logger logger = LogManager.getLogger();
+  public static final String TRANSACTIONS_TABLE = "transaction";
+
+  @Override
+  public Future<List<Transaction>> getTransactionsByCriterion(Criterion criterion, DBConn conn) {
+    logger.debug("Trying to get transactions by query: {}", criterion);
+    return conn.get(TRANSACTIONS_TABLE, Transaction.class, criterion)
+      .map(Results::getResults)
+      .onSuccess(transactions -> logger.info("Successfully retrieved {} transactions", transactions.size()))
+      .onFailure(e -> logger.error("Getting transactions failed", e));
+  }
+
+  @Override
+  public Future<List<Transaction>> getTransactionsByIds(List<String> ids, DBConn conn) {
+    logger.debug("Trying to get transactions by ids = {}", ids);
+    if (ids.isEmpty()) {
+      return Future.succeededFuture(Collections.emptyList());
+    }
+    CriterionBuilder criterionBuilder = new CriterionBuilder("OR");
+    ids.forEach(id -> criterionBuilder.with("id", id));
+    return getTransactionsByCriterion(criterionBuilder.build(), conn);
+  }
+
+  @Override
+  public Future<Void> createTransactions(List<Transaction> transactions, DBConn conn) {
+    return conn.saveBatch(TRANSACTIONS_TABLE, transactions)
+      .mapEmpty();
+  }
+
+  @Override
+  public Future<Void> updateTransactions(List<Transaction> transactions, DBConn conn) {
+    return conn.updateBatch(TRANSACTIONS_TABLE, transactions)
+      .mapEmpty();
+  }
+
+  @Override
+  public Future<Void> deleteTransactionsByIds(List<String> ids, DBConn conn) {
+    logger.debug("Trying to delete transactions by ids = {}", ids);
+    if (ids.isEmpty()) {
+      return Future.succeededFuture();
+    }
+    CriterionBuilder criterionBuilder = new CriterionBuilder("OR");
+    ids.forEach(id -> criterionBuilder.with("id", id));
+    return conn.delete(TRANSACTIONS_TABLE, criterionBuilder.build())
+      .onSuccess(transactions -> logger.info("Successfully deleted {} transactions", ids.size()))
+      .onFailure(e -> logger.error("Deleting transactions failed", e))
```

Comment:
error logs can be improved a bit, for example here by adding ids, in getTransactionsByCriterion add criterion to error log

Reply:
OK